src/signer/noscrypt_signer.cpp

```diff
@@ -29,14 +32,64 @@ static shared_ptr<NCContext> ncAllocContext()
     * be freed manually with the above helper when the smart 
     * pointer is destroyed
     */
-	void* ctxMemory = operator new(NCGetContextStructSize());
+    void* ctxMemory = operator new(NCGetContextStructSize());
 
-	return shared_ptr<NCContext>(
+    return shared_ptr<NCContext>(
         static_cast<NCContext*>(ctxMemory), 
         _ncFreeContext
     );
 }
 
+static shared_ptr<NCContext> initNoscryptContext()
+{
+    //Use helper to allocate a dynamic sized shared pointer
+    auto ctx = ncAllocContext();
+
+    auto randomEntropy = make_unique<uint8_t>(NC_CONTEXT_ENTROPY_SIZE);
+    NostrSecureRng::fill(randomEntropy.get(), NC_CONTEXT_ENTROPY_SIZE);
+
+    NCResult initResult = NCInitContext(ctx.get(), randomEntropy.get());
+
+    NC_LOG_ERROR(initResult);
+
+    return ctx;
+};
+
+/**
+ * @brief Generates a private/public key pair for local use.
+ * @returns The generated keypair of the form `[privateKey, publicKey]`, or a pair of empty
+ * strings if the function failed.
+ * @remarks This keypair is intended for temporary use, and should not be saved or used outside
+ * of this class.
+ */
+static void createLocalKeypair(
+    const shared_ptr<const NCContext> ctx,
+    shared_ptr<NCSecretKey> secret,
+    shared_ptr<NCPublicKey> pubkey
```

Comment:
Why modify these by side effect and rely on the caller to allocate the memory, rather than allocate the memory here and return the pair of `std::shared_ptr<NCSecretKey>` as a tuple?

I generally prefer operating by return value rather than by side effect, all else being equal.

Reply:
Would agree except when passing pointers, I think it's fair to assume they will be modified. That said I will follow your guidance, but I dislike tuples because I personally think they can get kind of messy, plus larger struct copies. Also it would cause multiple allocs which can be avoided when passing pointers.

Reply:
I should also mention, while I suppose C++ takes more of an OO approach to memory ownership I usually assume the memory owner will also be the one to allocate it. Noscrypt does not take on the role of managing memory for users so just a preference thing I suppose.

Reply:
Good points, I'm thinking those over.

On the multiple allocations, you're foreseeing an allocation at construction time in the signer, and a reallocation when the shared pointers are created?

C++ is a different beast from other languages I'm used to, so it may be better to simply allow the objects referenced by the pointers to be changed, as that may be a more language-native pattern.